src/server/scripts/Outland/zone_blades_edge_mountains.cpp

```diff
@@ -30,78 +30,178 @@ EndContentData */
     {
-        return new npc_deahts_door_wrap_gateAI(creature);
-    }
+        npc_deaths_door_fell_cannon_target_bunnyAI(Creature* creature) : ScriptedAI(creature), PartyTime(false) { }
 
-    struct npc_deahts_door_wrap_gateAI : public ScriptedAI
-    {
-        npc_deahts_door_wrap_gateAI(Creature *c) : ScriptedAI(c)
+        EventMap events;
+        bool PartyTime;
+        uint64 PlayerGUID;
+        uint64 CannonGUID;
+        uint8 count;
+        
+        void Reset() override
         {
+            Initialize();
+            events.Reset();
         }
 
-        uint8 count;
-        uint32 timer;
-        void Reset() { count = 0; timer = 0;}
-        void SpellHit(Unit* caster, SpellInfo const* spellInfo)
+        void Initialize()
         {
-            if (spellInfo->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
-            {
-                Player* plr = caster->GetCharmerOrOwnerPlayerOrPlayerItself();
-                if (!plr)
-                    return;
+            PartyTime = false;
+            PlayerGUID = 0;
+            CannonGUID = 0;
+            count = 0;
+        }
 
-                timer = 1;
+        void SpellHit(Unit* caster, SpellInfo const* spell) override
+        {
+            if (spell->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
+            {
                 count++;
 
+                if (count)
+                {
+                    if (Player* player = caster->GetCharmerOrOwnerPlayerOrPlayerItself())
+                        PlayerGUID = player->GetGUID();
+
+                    CannonGUID = caster->GetGUID();
+                    PartyTime = true;
+                    events.ScheduleEvent(EVENT_PARTY_TIMER, 3000);
+                }
+
+                if (count >= 3)
+                    me->SummonGameObject(GO_FIRE, me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 130);
+
                 if (count >= 6)
                 {
-                    if (me->GetEntry() == NPC_SOUTH_GATE)
-                        plr->KilledMonsterCredit(NPC_SOUTH_GATE_CREDIT, 0);
-                    else
-                        plr->KilledMonsterCredit(NPC_NORTH_GATE_CREDIT, 0);
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID))
+                    {
+                        if (GetClosestCreatureWithEntry(me, NPC_SOUTH_GATE, 200.0f))
+                            player->KilledMonsterCredit(NPC_SOUTH_GATE_CREDIT, TRIGGERED_NONE);
+                        else if (GetClosestCreatureWithEntry(me, NPC_NORTH_GATE, 200.0f))
+                            player->KilledMonsterCredit(NPC_NORTH_GATE_CREDIT, TRIGGERED_NONE);
+                        // complete quest part
+                        if (Creature* bunny = GetClosestCreatureWithEntry(me, NPC_EXPLOSION_BUNNY, 200.0f))
+                            bunny->CastSpell(nullptr, SPELL_EXPLOSION, TRIGGERED_NONE);
+                    }
 
```

Comment:
Please add this in order to despawn the cannon after the warp gate is destroyed:
```cpp
                    if (Creature* cannon = ObjectAccessor::GetCreature(*me, CannonGUID))
                        cannon->DespawnOrUnsummon(5000);
```